inbm/dispatcher-agent/tests/unit/source/test_ubuntu_source_cmd.py

```diff
@@ -279,6 +279,19 @@ def test_successfully_remove_gpg_key_and_source_list(
         except SourceError:
             self.fail("Remove GPG key raised DispatcherException unexpectedly!")
 
+     @patch("dispatcher.packagemanager.package_manager.verify_source", side_effect=DispatcherException('error'))
+     def test_successfully_add_gpg_key(self, mock_verify_source):
```

Comment:
The test case name `test_successfully_add_gpg_key` is misleading since it seems to test the failure scenario. Rename it to reflect the actual test purpose. [medium]